### PR TITLE
add discussion for string formatting

### LINKED ISSUE #83: Link to this great resource for formatting strings
https://pyformat.info/

### PR DESCRIPTION
**Pull request by rsokl**
@davidmascharka I'm not super happy with my treatment of f-strings. Please feel free to spruce this material up if you see any low-hanging fruit.

Closes #83

### CODE REVIEW

File: Python/Module2_EssentialsOfPython/Basic_Objects.md [hunk 1]
```diff
@@ -721,11 +721,77 @@ True
...
+```python
+# this is a typical empty string
+>>> ""
+''
+
+# this is an empty f-string
+>>> f""
+''
+```
+
```
Comment (davidmascharka) [inline Python/Module2_EssentialsOfPython/Basic_Objects.md]: This is your call but I don't think I'd put the brackets inside parens.

"within curly brackets (`{}`) will be"

"within curly brackets `{}` will be"

....I kind of don't like either. Maybe

"within curly brackets (e.g. `{x - 3}`) will be"

or something.

File: Python/Module2_EssentialsOfPython/Basic_Objects.md [hunk 2]
```diff
@@ -721,11 +721,77 @@ True
...
+```python
+# using `format` to replace placeholders with values
+>>> name = "Bruce"
+>>> age = 80
+>>> "{name} is {age} years old".format(name="Bruce", age=80)
+'Bruce is 80 years old'
+
+# padding a string with 4 leading spaces
+>>> "{item:>4}".format(item="stew")
+'   stew'
```
Comment (davidmascharka) [inline Python/Module2_EssentialsOfPython/Basic_Objects.md]: This is not correct.

```python
In [1]: '{item:>4}'.format(item='stew')                                                        
Out[1]: 'stew'
```

You probably want `item:>8`

Comment (rsokl) [reply Python/Module2_EssentialsOfPython/Basic_Objects.md]: Whew! Good catch!

File: Python/Module2_EssentialsOfPython/Basic_Objects.md (hunk 1 above)
Comment (rsokl) [reply Python/Module2_EssentialsOfPython/Basic_Objects.md]: Good call

**Pull request merged**